crates/sui-indexer/src/balance_changes.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::ops::Neg;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use move_core_types::language_storage::TypeTag;
+use tokio::sync::RwLock;
+
+use sui_json_rpc_types::BalanceChange;
+use sui_json_rpc_types::{SuiExecutionStatus, SuiTransactionEffects, SuiTransactionEffectsAPI};
+use sui_types::base_types::{MoveObjectType, ObjectID, SequenceNumber};
+use sui_types::coin::Coin;
+use sui_types::gas::GasCostSummary;
+use sui_types::gas_coin::GAS;
+use sui_types::object::{Object, Owner};
+
+use crate::errors::IndexerError;
+use crate::store::PgIndexerStore;
+
+pub async fn get_balance_changes_from_effect<P: ObjectProvider<Error = E>, E>(
+    object_provider: &P,
+    effects: &SuiTransactionEffects,
+) -> Result<Vec<BalanceChange>, E> {
+    let gas_owner = effects.gas_object().owner;
+
+    // Only charge gas when tx fails, skip all object parsing
+    let sui_gas_cost_summary = effects.gas_cost_summary();
+    let gas_cost_summary = GasCostSummary {
```

Comment:
doesn't look like we use `sui_gas_cost_summary` or `gas_cost_summary` other than in L37-40, are you not able to do `amount: effects.get_cost_summary()` directly

Reply:
`effects.get_cost_summary()` is the JSON RPC type, we need the `sui-core` type, so what I was doing was just converting them, added an into instead.